internal/framework/generic_resource.go

```diff
@@ -310,13 +313,34 @@ func (r *GenericResource) GetTimeout(operation string, defaultTimeout time.Durat
 	return defaultTimeout
 }
 
+type resourceFailed struct {
+	message string
+}
+
+func ResourceFailed(format string, args ...any) error {
+	return &resourceFailed{message: fmt.Sprintf(format, args...)}
+}
+
+func (err *resourceFailed) Error() string {
```

Comment:
This implements the `Error` interface. Can you please enforce that? E.g.

```
var _ error = &resourceFailedError{}
```

Reply:
Will do.

Reply:
The function on line 320 serves the same purpose.

Reply:
Right. Please disregard then.